src/extensions/averagewaveforms/AverageWaveformsView/setupElectrodes.ts

```diff
@@ -44,8 +44,61 @@ const getElectrodesBoundingBox = (electrodeLocations: number[][], radius: number
     }
 }
 
-const setupElectrodes = (args: {width: number, height: number, electrodeLocations: Vec2[], electrodeIds: number[]}) => {
-    const { width, height, electrodeLocations, electrodeIds } = args
+export const getElectrodesAspectRatio = (electrodeLocations: Vec2[]) => {
+    const radius = computeRadius(electrodeLocations)
+    let boundingBox = getElectrodesBoundingBox(electrodeLocations, radius)
+    let boxAspect = getWidth(boundingBox) / getHeight(boundingBox)
+    return boxAspect
+}
+
+const setupVerticalElectrodes = ({width, height, electrodeIds}: {width: number, height: number, electrodeIds: number[]}) => {
+    const xMargin = 10
+    const yMargin = 10
+    const n = electrodeIds.length
+    
+    const transform = funcToTransform((p: Vec2): Vec2 => {
+        const x = xMargin + p[0] * ( width - 2 * xMargin )
+        const y = yMargin + p[1] * ( height - 2 * yMargin )
+        return [x, y]
+    })
+
+    const electrodeBoxes: ElectrodeBox[] = electrodeIds.map((eid, ii) => {
+        const y = (0.5 + ii) / (n + 1)
+        const rect = {xmin: 0, xmax: 1, ymin: y - 0.5 / (n + 1), ymax: y + 0.5 / (n + 1)}
+        const transform0 = funcToTransform((p: Vec2): Vec2 => {
+            const a = rect.xmin + p[0] * (rect.xmax - rect.xmin)
+            const b = rect.ymin + p[1] * (rect.ymax - rect.ymin)
+            return [a, b]
+        })
+        return {
+            label: eid + '',
+            id: eid,
+            x: 0.5,
+            y,
+            rect,
+            transform: transform0
+        }
+    })
```

Comment:
This function is trying to create bounding canvas areas for each waveform for the vertical display? With each channel mapped to its own space, whose width is the full width of the unit and whose height is 1/nth of it, less a margin?

The screenshot shows it works, but it's not super easy to follow what it does, nor match up the behavior to confirm that it does what's intended. We should add some comments here (at least l. 65-81).